### PR TITLE
Add makefile

### LINKED ISSUE #213: Create a Makefile with local development commands
**Please describe the feature you'd like to see**
At the moment, we expect developers to set up a development environment by manually following the steps described in the document [DEVELOPMENT.md](https://github.com/astro-projects/astro/blob/main/docs/DEVELOPMENT.md). This process is prone to errors. The goal with this ticket is to automate those steps where possible, by having a project `Makefile`.

**Describe the solution you'd like**
Features:
- [x] `make setup`: to create a Python virtual environment and install all the dependencies necessary to this local environment (including Airflow). It should also install pre-commit and download the Postgres docker image. Make a copy of the configuration files (`.env` and `ci-test-connections.yml`), if they don't exist, and prompt the user to update them with their credentials.
- [x] `make tests`: to run all the tests and include coverage using `--cov-branch and --cov-report term-missing`, so the developer can see which lines haven't been covered.
- [x] `make unit_test`: to run all the unit tests, which don't depend on external databases or remote object stores (+ coverage)
- [x] `make integration_tests`: to run all the integration tests, which rely on external databases and/or remote object stores (+ coverage)
- [x] `make snowflake_tests`: to run all the integration tests which use Snowflake and none of the other DBs (+coverage)
- [x] `make postgres_tests`: to run all the integration tests which use Postgres and none of the other DBs (+coverage)
- [x] `make aws_tests`: to run all the integration tests which use S3 and/or Redshift and none of the other DBs (+coverage)
- [x] `make google_tests`: to run all the integration tests which use GCS and/or BigQuery and none of the other DBs (+coverage)
- [x] `make quality`: to run all the quality checks declared within `pre-commit`

**Acceptance Criteria**

- [ ] Have a Makefile that can be run in MacOS and Ubuntu which implements the following features

### PR DESCRIPTION
**Pull request by bhavaniravi**
Fixes #213 
Adds Makefile to ease project setup

**To-Do**
1. Differentiate integration and unit-test
2. Better documentation of commands
3. Extensive Testing

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	$(PIP) install pipx nox
+	$(PIP) install '.[$(if $(collection),$(collection),all)]'
+	$(PIP) install .[tests]
+
+activate: 
+	. $(VENV)/bin/activate
+	
+init_venv:
+	test -d venv || $(PYTHON) -m venv $(VENV)
+	. $(VENV)/bin/activate
```
Comment (dimberman) [inline Makefile:34]: Should venv be an env variable? Can we make it an argument?

e.g. `make activate --venv dev`

**Review by dimberman — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+test:
+ifdef db
+	pytest -s --cov-report term --cov-branch -m "$(db)"
+else
+	pytest -s --cov-report term --cov-branch
+endif
+
+unit_test:
+	pytest -s --cov-report term --cov-branch -m "not integration"
```
Comment (tatiana) [inline Makefile:45]: What do you think about using  `--cov-report term-missing` so we can see any implicit code branches that the tests might not be covering?

**Review by tatiana — APPROVED** [review]
It's a great beginning, @bhavaniravi , thank you very much!
Do you have any ideas about how we can improve it in the future, to handle the `.env` and `test-connections.yaml` files?

**Comment by pankajastro** [comment]
LGTM

**Review by pankajastro — APPROVED** [review]

**Review by dimberman — APPROVED** [review]

**Pull request merged**